lib/src/lints/named_parameters_ordering/named_parameters_ordering_rule.dart

```diff
@@ -0,0 +1,180 @@
+///   final String email;
+///   final bool isActive;
+///   final String name;
+///
+///   UserProfile({
+///     this.age,
+///     required super.accountType, // LINT, required super named parameters should be before nullable named parameters
+///     required this.name, // LINT, required named parameters should be before super named parameters
+///     super.userId,
+///     this.country, // LINT, nullable named parameters should be before super named parameters
+///     this.isActive = true, // LINT, default named parameters should be before nullable named parameters
+///     required this.email, // LINT, required named parameters should be before default named parameters
+///   });
+///
+///   void doSomething({
+///     required String name,
+///     int? age,
+///     bool isActive = true, // LINT, default named parameters should be before nullable named parameters
+///     required String email, // LINT, required named parameters should be before default named parameters
+///   }) {
+///     return;
+///   }
+/// }
+/// ```
+///
+/// #### GOOD:
+///
+/// ```dart
+/// class User {
+///   final String accountType;
+///   final String? userId;
+///
+///   User({
+///     required this.accountType,
+///     this.userId,
+///   });
+/// }
+///
+/// class UserProfile extends User {
+///   final String? age;
+///   final String? country;
+///   final String email;
+///   final bool isActive;
+///   final String name;
+///
+///   UserProfile({
+///     required this.name,
+///     required this.email,
+///     required super.accountType,
+///     this.isActive = true,
+///     this.age,
+///     this.country,
+///     super.userId,
+///   });
+///
+///   void doSomething({
+///     required String name,
+///     required String email,
+///     bool isActive = true,
+///     int? age,
+///   }) {
+///     return;
+///   }
+/// }
+/// ```
+class NamedParametersOrderingRule
+    extends SolidLintRule<NamedParametersOrderingParameters> {
+  /// This lint rule represents
+  /// the error whether we use bad formatted double literals.
+  static const lintName = 'named_parameters_ordering';
+
+  static const _warningMessage = 'should be before';
+
+  NamedParametersOrderingRule._(super.config);
+
+  /// Creates a new instance of [NamedParametersOrderingRule]
+  /// based on the lint configuration.
+  factory NamedParametersOrderingRule.createRule(CustomLintConfigs configs) {
+    final config = RuleConfig<NamedParametersOrderingParameters>(
+      configs: configs,
+      name: lintName,
+      paramsParser: NamedParametersOrderingParameters.fromJson,
+      problemMessage: (_) => "Order of named parameter is wrong",
+    );
+
+    return NamedParametersOrderingRule._(config);
+  }
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addFormalParameterList((node) {
+      final visitor = NamedParametersOrderingVisitor(config.parameters.order);
```

Comment:
Can we define this visitor on a class level to avoid re-creation each time this callback is called?